cache.go

```diff
@@ -0,0 +1,223 @@
+	"github.com/bitrise-io/go-utils/pathutil"
+	glob "github.com/ryanuber/go-glob"
+)
+
+// parseIncludeListItem separates path to cache and change indicator path.
+func parseIncludeListItem(item string) (string, string) {
+	// file/or/dir/to/cache -> indicator/file
+	// file/or/dir/to/cache
+	i := strings.Index(item, "->")
+	if i == -1 {
+		return strings.TrimSpace(item), ""
+	}
+	var pth string
+	if i > 0 {
+		pth = item[:i]
+	}
+
+	var indicator string
+	if i+2 < len(item) {
+		indicator = item[i+2:]
+	}
+
+	return strings.TrimSpace(pth), strings.TrimSpace(indicator)
+}
+
+// parseIgnoreListItem separates ignore pattern and if pattern match removes item from cache or not.
+func parseIgnoreListItem(item string) (string, bool) {
+	// path/or/patter/to/exclude
+	// !path/or/patter/to/exclude
+	item = strings.TrimSpace(item)
+	if len(item) == 0 {
+		return "", false
+	}
+
+	ignore := false
+	if strings.HasPrefix(item, "!") {
+		ignore = true
+		item = strings.TrimPrefix(item, "!")
+		item = strings.TrimSpace(item)
+		if len(item) == 0 {
+			return "", false
+		}
+	}
+
+	return item, ignore
+}
+
+func parseIncludeList(list []string) map[string]string {
+	indicatorByPath := map[string]string{}
+	for _, item := range list {
+		pth, indicator := parseIncludeListItem(item)
+		if len(pth) == 0 {
+			continue
+		}
+		indicatorByPath[pth] = indicator
+	}
+	return indicatorByPath
+}
+
+func parseIgnoreList(list []string) map[string]bool {
+	ignoreByPath := map[string]bool{}
+	for _, item := range list {
+		pth, ignore := parseIgnoreListItem(item)
+		if len(pth) == 0 {
+			continue
+		}
+		ignoreByPath[pth] = ignore
+	}
+	return ignoreByPath
+}
+
+// expandPath returns every file included in pth (recursively) if it is a dir,
+// if pth is a file it will be returned as an array.
+func expandPath(pth string) ([]string, error) {
+	info, err := os.Lstat(pth)
+	if err != nil {
+		return nil, err
+	}
+
+	var subPaths []string
+	if info.IsDir() {
+		if err := filepath.Walk(pth, func(p string, i os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if i.IsDir() {
+				return nil
+			}
+
+			subPaths = append(subPaths, p)
+			return nil
+		}); err != nil {
+			return nil, err
+		}
+	} else {
+		subPaths = append(subPaths, pth)
+	}
+
+	return subPaths, nil
+}
+
+// normalizeIndicatorByPath modifies indicatorByPath:
+// expands both path to cache and indicator path
+// removes the item if any of path to cache or indicator path is not exist or if the indicator is a dir
+// replaces path to cache (if it is a directory) by every file (recursively) in the directory.
+func normalizeIndicatorByPath(indicatorByPath map[string]string) (map[string]string, error) {
+	normalized := map[string]string{}
+	for pth, indicator := range indicatorByPath {
+		if len(indicator) > 0 {
+			var err error
+			indicator, err = pathutil.AbsPath(indicator)
+			if err != nil {
+				return nil, err
+			}
+
+			info, exist, err := pathutil.PathCheckAndInfos(indicator)
```

Comment:
you can move this line in the switch without expression (using a single `;`)
for example:
```
switch info, exist, err := pathutil.PathCheckAndInfos(indicator); {
```

https://yourbasic.org/golang/switch-statement

Reply:
wow!